packages-internal/test-utils/src/describeConformance.tsx

```diff
@@ -134,16 +96,20 @@ export function testClassName(
   getOptions: () => ConformanceOptions,
 ) {
   it('applies the className to the root component', () => {
-    const { mount } = getOptions();
-    if (!mount) {
-      throwMissingPropError('mount');
+    const { render } = getOptions();
+
+    if (!render) {
+      throwMissingPropError('render');
     }
 
     const className = randomStringValue();
+    const testId = randomStringValue();
 
-    const wrapper = mount(React.cloneElement(element, { className }));
+    const { getByTestId } = render(
+      React.cloneElement(element, { className, 'data-testid': testId }),
```

Comment:
This relies on the component applying `className` in the same component that `data-testid` is applied. This is followed in most components, that component being the root. We have 2 exceptions (Material's Switch and Joy's Tooltip), for which I skipped the test and added an explanation.